tests/test_resolver.py

```diff
@@ -187,6 +184,14 @@ def test_resolve_dependencies__generates_expected_mappings(
         install_deps=install_deps,
     )
 
+    # patch TemporaryPipInstallResolver so that it's called with the cache
+    # option in the resolver
+    tmp_pip_install_resolver = TemporaryPipInstallResolver(cache=request.config.cache)
+    mocker.patch(
+        "fawltydeps.packages.TemporaryPipInstallResolver",
+        return_value=tmp_pip_install_resolver,
+    )
+
```

Comment:
Here is where I'd like to place the code that "knows" about the pytest cache. We turn that into a simple `Path` that we can store into `TemporaryPipInstallResolver.cached_venv`.

Additionally, since I'm not 100% sure how changing the `TemporaryPipInstallResolver` class will work across various test cases, I want to _reset_ the class member back to `None` after the test is run, with a `try ... finally` block. After all, we _don't_ want a `TemporaryPipInstallResolver.cached_venv` from this test to affect the `TemporaryPipInstallResolver` unit tests!

```python
    # Tell TemporaryPipInstallResolver to reuse our cached venv, instead of
    # potentially creating a new venv for every test case.
    cached_venv = Path(
        request.config.cache.mkdir(
            f"fawltydeps_reused_venv_{sys.version_info.major}.{sys.version_info.minor}"
        )
    )
    try:
        TemporaryPipInstallResolver.cached_venv = cached_venv

        obtained = resolve_dependencies(
            ...
        )
    finally:
        TemporaryPipInstallResolver.cached_venv = None

    assert obtained == expected
```